src/snowflake/connector/local_storage_client.py

```diff
@@ -55,7 +55,7 @@ def download_chunk(self, chunk_id: int) -> None:
                 os.path.join(
                     self.meta.local_location, os.path.basename(self.meta.dst_file_name)
                 ),
-                "wb",
```

Comment:
Is this because we wanna append instead of truncate?

Reply:
Not really append. We want to open the file for update. See: https://docs.python.org/3/library/functions.html#open

> The default mode is 'r' (open for reading text, a synonym of 'rt'). Modes 'w+' and 'w+b' open and truncate the file. Modes 'r+' and 'r+b' open the file with no truncation.

Reply:
We are using `rb+` for other storage client in https://github.com/snowflakedb/snowflake-connector-python/blob/ea8d733752b2efbcbe712fe4cd9a47cce0f88776/src/snowflake/connector/storage_client.py#L354